app/models/custom_filter.rb

```diff
@@ -24,6 +24,28 @@ class CustomFilter < ApplicationRecord
   enum filter_type: { conversation: 0, contact: 1, report: 2 }
   validate :validate_number_of_filters
 
+  def records_count
+    fetch_record_count_from_redis || set_record_count_in_redis
```

Comment:
Currently, some of the filters take quite a long while. so calculating it on the fly for each filter might time out the request.

If the `key` is nil lets return `nil` and enqueue the job to update the key rather. 

So that the value will be available in subsequent requests.